kotlin-spark-api/3.2/src/test/kotlin/org/jetbrains/kotlinx/spark/api/ApiTest.kt

```diff
@@ -318,24 +324,97 @@ class ApiTest : ShouldSpec({
                 cogrouped.count() shouldBe 4
             }
             should("handle LocalDate Datasets") { // uses encoder
-                val dataset: Dataset<LocalDate> = dsOf(LocalDate.now(), LocalDate.now())
-                dataset.show()
+                val dates = listOf(LocalDate.now(), LocalDate.now())
+                val dataset: Dataset<LocalDate> = dates.toDS()
+                dataset.collectAsList() shouldBe dates
             }
             should("handle Instant Datasets") { // uses encoder
-                val dataset: Dataset<Instant> = dsOf(Instant.now(), Instant.now())
-                dataset.show()
+                val instants = listOf(Instant.now(), Instant.now())
+                val dataset: Dataset<Instant> = instants.toDS()
+                dataset.collectAsList() shouldBe instants
+            }
+            should("Be able to serialize Instant") { // uses knownDataTypes
+                val instantPair = Instant.now() to Instant.now()
+                val dataset = dsOf(instantPair)
+                dataset.collectAsList() shouldBe listOf(instantPair)
             }
             should("be able to serialize Date") { // uses knownDataTypes
-                val dataset: Dataset<Pair<Date, Int>> = dsOf(Date.valueOf("2020-02-10") to 5)
-                dataset.show()
+                val datePair = Date.valueOf("2020-02-10") to 5
+                val dataset: Dataset<Pair<Date, Int>> = dsOf(datePair)
+                dataset.collectAsList() shouldBe listOf(datePair)
             }
             should("handle Timestamp Datasets") { // uses encoder
-                val dataset = dsOf(Timestamp(0L))
-                dataset.show()
+                val timeStamps = listOf(Timestamp(0L), Timestamp(1L))
+                val dataset = timeStamps.toDS()
+                dataset.collectAsList() shouldBe timeStamps
             }
             should("be able to serialize Timestamp") { // uses knownDataTypes
-                val dataset = dsOf(Timestamp(0L) to 2)
-                dataset.show()
+                val timestampPair = Timestamp(0L) to 2
+                val dataset = dsOf(timestampPair)
+                dataset.collectAsList() shouldBe listOf(timestampPair)
+            }
+            should("handle Duration Datasets") { // uses encoder
+                val dataset = dsOf(Duration.ZERO)
+                dataset.collectAsList() shouldBe listOf(Duration.ZERO)
+            }
+            should("handle Period Datasets") { // uses encoder
+                val periods = listOf(Period.ZERO, Period.ofDays(2))
+                val dataset = periods.toDS()
+
+                dataset.show(false)
+
+                dataset.collectAsList().let {
+                    it[0] shouldBe Period.ZERO
+
+                    // TODO this is also broken in Scala. It reports a Period of 0 instead of 2 days
```

Comment:
Please link relevant JIRA ticket

Reply:
https://issues.apache.org/jira/browse/SPARK-38317 Apparently it's expected behavior, strange